docs/configuration.md

```diff
@@ -954,12 +971,12 @@ Apart from these, the following properties are also available, and may be useful
   <td><code>spark.memory.offHeap.enabled</code></td>
   <td>false</td>
   <td>
-    If true, Spark will attempt to use off-heap memory for certain operations. If off-heap memory use is enabled, then <code>spark.memory.offHeap.size</code> must be positive.
+    If true, Spark will attempt to use off-heap memory for certain operations(e.g. sort, aggregate, etc. However, the buffer used for fetching shuffle blocks is always off-heap). If off-heap memory use is enabled, then <code>spark.memory.offHeap.size</code> must be positive.
```

Comment:
Isn't 

> the buffer used for fetching shuffle blocks is always off-heap

contradicted by the `preferDirectBufs()` setting being configurable?